contracts/curve/CurveMath6.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.13;
+
+import "../number/types/UFixed6.sol";
+import "../number/types/Fixed6.sol";
+
+/**
+ * @title CurveMath6
+ * @notice Library for managing math operations for utilization curves.
+ */
+library CurveMath6 {
+    error CurveMath6OutOfBoundsError();
+
+    /**
+     * @notice Computes a linear interpolation between two points
+     * @param startX First point's x-coordinate
+     * @param startY First point's y-coordinate
+     * @param endX Second point's x-coordinate
+     * @param endY Second point's y-coordinate
+     * @param targetX x-coordinate to interpolate
+     * @return y-coordinate for `targetX` along the line from (`startX`, `startY`) -> (`endX`, `endY`)
+     */
+    function linearInterpolation(
+        UFixed6 startX,
+        Fixed6 startY,
+        UFixed6 endX,
+        Fixed6 endY,
+        UFixed6 targetX
+    ) internal pure returns (Fixed6) {
+        if (targetX.lt(startX) || targetX.gt(endX)) revert CurveMath6OutOfBoundsError();
+
+        UFixed6 xRange = endX.sub(startX);
+        Fixed6 yRange = endY.sub(startY);
+        UFixed6 xRatio = targetX.sub(startX).div(xRange);
+        return yRange.mul(Fixed6Lib.from(xRatio)).add(startY);
+    }
+
+    /**
+     * @notice Computes a linear interpolation between two points
+     * @param startX First point's x-coordinate
+     * @param startY First point's y-coordinate
+     * @param endX Second point's x-coordinate
+     * @param endY Second point's y-coordinate
+     * @param targetX x-coordinate to interpolate
+     * @return y-coordinate for `targetX` along the line from (`startX`, `startY`) -> (`endX`, `endY`)
+     */
+    function linearInterpolation(
+        UFixed6 startX,
+        UFixed6 startY,
+        UFixed6 endX,
+        UFixed6 endY,
+        UFixed6 targetX
+    ) internal pure returns (UFixed6) {
+        if (targetX.lt(startX) || targetX.gt(endX)) revert CurveMath6OutOfBoundsError();
+
+        UFixed6 xRange = endX.sub(startX);
+        (UFixed6 yRange, bool addToStartY) = endY.gte(startY) ? (endY.sub(startY), true) : (startY.sub(endY), false);
```

Comment:
good catch on this case, I'm thinking it's simpler to just replace this function implementation with something like this:

```
        return UFixed6Lib.from(linearInterpolation(startX, Fixed6Lib.from(startY), endX, Fixed6Lib.from(endY), targetX));
```

Reply:
ok, so we can assume that `startY` and `endY` will never be out of bounds on `Fixed6`?